munch-match/src/components/resultsLayout.jsx

```diff
@@ -1,105 +1,157 @@
+                const sortedResults = results.sort((r1, r2) => r2.rating - r1.rating);
+                this.setState({results: sortedResults});
+                // Get the coordinates of the restaurants and initialize our embedded map
+                const locations = results.map(result => {
+                    return {
+                        latitude: result.latitude,
+                        longitude: result.longitude
+                    };
+                });
+                this.initializeMap(locations);
+            })
+            .catch(error => {
+                console.error('Error fetching results:', error);
             });
-            this.initializeMap(locations); 
-          })
-          .catch(error => {
-            console.error('Error fetching results:', error);
-          });
-      }
-      // Function to initialize embedded map
-      initializeMap = (foodLocations) => {
+    }
+    // Function to initialize embedded map
+    initializeMap = (foodLocations) => {
         const center = {lat: foodLocations[0].latitude, lng: foodLocations[0].longitude}
         const map = new google.maps.Map(document.getElementById('map'), {
-          zoom: 13, 
-          center: center
+            zoom: 13,
+            center: center
         });
 
         // Loop through each location and extract its coordinates and make a new marker on the map
         foodLocations.forEach(location => {
             new google.maps.Marker({
-            position: {lat: location.latitude, lng: location.longitude}, 
-            map: map
-          })
+                position: {lat: location.latitude, lng: location.longitude},
+                map: map
+            })
         })
-      }
-
-    render(){
-        console.log(this.selectedType)
-        const { results } = this.state;
-        console.log(results[2]);
-
-        return(
-        //To export multiple components, surround it with a <div> tag
-        <div className={styles.resultsContainer}>
-            <div className={styles.resultsRow}>
-                <ResultCard result={results[0]}/>
-                <ResultCard result={results[1]}/>
-                <ResultCard result={results[2]}/>
-            </div>
-            <div className={styles.resultsRow}>
-                <ResultCard result={results[3]}/>
-                <ResultCard result={results[4]}/>
-                <ResultCard result={results[5]}/>
-            </div>
-            <div id="map" className={styles.map}></div>
-            <div>
-            <button className={styles.buttonResultLayout} onClick={() => this.onBackClick()} style={{width: "30%"}}><h3>Choose Again</h3></button>
+    }
+
+    updateRating = (range) => {
+        this.setState({ratingRange: range});
+        this.componentDidMount();
+    }
+
+    updatePriceRange = (price) => {
+        const newPriceRanges = { ...this.state.priceRanges };
+        newPriceRanges[price] = !newPriceRanges[price];
```

Comment:
You could consider updating the state in a way that doesn't directly modify it - this it safer and helps prevent bugs down the line. Instead of altering newPriceRanges directly, you could use setState to create a new copy of priceRanges with the changes. So something like this:

`this.setState(prevState => ({
  priceRanges: { ...prevState.priceRanges, [price]: !prevState.priceRanges[price] }
}));`

again this makes it so that we're not directly altering the original state, and this is also a react best practice